packages/typography/src/elements/CodeBlock.tsx

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { HTMLAttributes } from 'react';
+import Highlight, { Language, Prism } from 'prism-react-renderer';
+import { StyledCodeBlock, StyledCodeBlockLine, StyledCodeBlockToken } from '../styled';
+
+export interface ICodeBlockProps extends HTMLAttributes<HTMLPreElement> {
+  language?: Language;
+  size?: 'small' | 'medium' | 'large';
+  isLight?: boolean;
+  isNumbered?: boolean;
+}
+
+/**
+ * @extends HTMLAttributes<HTMLPreElement>
+ */
+const CodeBlock: React.FunctionComponent<
+  ICodeBlockProps & React.RefAttributes<HTMLPreElement>
+> = React.forwardRef(({ children, isLight, isNumbered, language, size, ...other }, ref) => {
```

Comment:
`React.forwardRef<T, P>()` takes two generics which can remove the need for `React.RefAttributes` and `React.FunctionComponent`.

Reply:
At present, it seems we have dozens of declarations scattered throughout the codebase that don't line up with the simplified syntax. Would probably be good to set a task to clean those up so folks (like me 😅 ) don't keep following suboptimal examples.